tests/layer/test_link_inference.py

```diff
@@ -25,6 +25,140 @@
 import pytest
 
 
+class Test_LinkEmbedding(object):
+    """
+    Group of tests for link_inference() function
+    """
+
+    d = 100  # dimensionality of embedding vector space
+    d_out = 10  # dimensionality of link inference output
+
+    def test_ip(self):
+        """ Test the 'ip' binary operator on orthogonal vectors"""
+
+        x_src = np.random.randn(self.d)
+        x_src /= np.linalg.norm(x_src)  # normalize x_src
+        x_dst = np.random.randn(self.d)
+        x_dst -= x_dst.dot(x_src) * x_src  # make x_dst orthogonal to x_src
+        x_dst /= np.linalg.norm(x_dst)  # normalize x_dst
+
+        expected = np.dot(x_src, x_dst)
+
+        sess = tf.InteractiveSession()
+
+        x_src = tf.constant(x_src, shape=(1, self.d), dtype="float64")
+        x_dst = tf.constant(x_dst, shape=(1, self.d), dtype="float64")
+
+        li = LinkEmbedding(method="ip", activation="linear")([x_src, x_dst])
+        print(
+            "link inference with 'ip' operator on orthonormal vectors: {}, expected: {}".format(
+                li.eval(), expected
+            )
+        )
+        assert li.eval() == pytest.approx(expected)
+        assert li.eval() == pytest.approx(0)
+
+        li = LinkEmbedding(method="ip", activation="linear")([x_src, x_src])
+        print("link inference with 'ip' operator on unit vector: ", li.eval())
+        assert li.eval() == pytest.approx(1)
+
+        # Test sigmoid activation
+        li = LinkEmbedding(method="ip", activation="sigmoid")([x_src, x_dst])
+        assert li.eval() == pytest.approx(0.5)
+
+        li = LinkEmbedding(method="ip", activation="sigmoid")([x_src, x_src])
+        assert li.eval() == pytest.approx(0.7310586)
+
+        sess.close()
+
+    def test_ip_single_tensor(self):
+        """ Test the 'ip' binary operator on orthogonal vectors"""
+
+        x_src = np.random.randn(self.d)
+        x_src /= np.linalg.norm(x_src)  # normalize x_src
+        x_dst = np.random.randn(self.d)
+        x_dst -= x_dst.dot(x_src) * x_src  # make x_dst orthogonal to x_src
+        x_dst /= np.linalg.norm(x_dst)  # normalize x_dst
+
+        expected = np.dot(x_src, x_dst)
+
+        sess = tf.InteractiveSession()
+
+        x_src = tf.constant(x_src, shape=(1, self.d), dtype="float64")
+        x_dst = tf.constant(x_dst, shape=(1, self.d), dtype="float64")
+        x_link_sd = tf.stack([x_src, x_dst], axis=1)
+        x_link_ss = tf.stack([x_src, x_src], axis=1)
+
+        li = LinkEmbedding(method="ip", activation="linear")(x_link_sd)
+        print(
+            "link inference with 'ip' operator on orthonormal vectors: {}, expected: {}".format(
+                li.eval(), expected
+            )
+        )
+        assert li.eval() == pytest.approx(expected)
+        assert li.eval() == pytest.approx(0)
```

Comment:
```suggestion
        assert expected == pytest.approx(0)
```
Perhaps put this assertion up above with the computation of `expected`?